app/services/LocationService.js

```diff
@@ -383,70 +423,93 @@ export default class LocationServices {
-        //     ),
-        //   1000,
-        // );
+        //we need to set delay or otherwise alert may not be shown
+        setTimeout(
+          () =>
+            Alert.alert(
+              languages.t('label.require_location_services_title'),
+              languages.t('label.require_location_services_message'),
+              [
+                {
+                  text: languages.t('label.yes'),
+                  onPress: () => {
+                    if (isPlatformAndroid()) {
+                      // showLocationSettings() only works for android
+                      BackgroundGeolocation.showLocationSettings();
+                    } else {
+                      // eslint-disable-next-line
+                      Linking.openURL('App-Prefs:Privacy'); // Deeplinking method for iOS
+                    }
+                  },
+                },
+                {
+                  text: languages.t('label.no'),
+                  onPress: () => {
+                    console.log('No Pressed'), LocationServices.stop();
+                  },
+                  style: 'cancel',
+                },
+              ],
+            ),
+          1000,
+        );
       } else if (!status.authorization) {
         // we need to set delay or otherwise alert may not be shown
-        // setTimeout(
-        //   () =>
-        //     Alert.alert(
-        //       languages.t('label.require_location_information_title'),
-        //       languages.t('label.require_location_information_message'),
-        //       [
-        //         {
-        //           text: languages.t('label.yes'),
-        //           onPress: () => BackgroundGeolocation.showAppSettings(),
-        //         },
-        //         {
-        //           text: languages.t('label.no'),
-        //           onPress: () => console.log('No Pressed'),
-        //           style: 'cancel',
-        //         },
-        //       ],
-        //     ),
-        //   1000,
-        // );
+        setTimeout(
+          () =>
+            Alert.alert(
+              languages.t('label.require_location_information_title'),
+              languages.t('label.require_location_information_message'),
+              [
+                {
+                  text: languages.t('label.yes'),
+                  onPress: () => BackgroundGeolocation.showAppSettings(),
+                },
+                {
+                  text: languages.t('label.no'),
+                  onPress: () => {
+                    console.log('No Pressed'), LocationServices.stop();
+                    //                     Settings.setIsLogging(false);
+                  },
+                  style: 'cancel',
+                },
+              ],
+            ),
+          1000,
+        );
       }
     });
   }
 
+  static isheadlessAlreadyRunning() {
+    return BackgroundGeolocation.getConfig(config => {
+      if (!config.stopOnTerminate && config.notificationsEnabled) return true;
```

Comment:
this could be simplified to 

```js
return BackgroundGeolocation.getConfig(config => {
  return !config.stopOnTerminate && config.notificationsEnabled
});
```

Reply:
The return type of this method is probably always `undefined` because [it's not a promise](https://github.com/mauron85/react-native-background-geolocation/blob/c13b3ce45a16a1b538be6ddb1bb02f48c23354fb/index.js#L130-L134), can you confirm?

I think what you want is:

```js
return new Promise((resolve, reject) => {
  BackgroundGeolocation.getConfig((config, error) => {
    if (error) { 
      reject(error);
    } else {
      resolve(!config.stopOnTerminate && config.notificationsEnabled);
    }
});
```